be/src/olap/txn_manager.cpp

```diff
@@ -514,6 +553,21 @@ Status TxnManager::publish_txn(OlapMeta* meta, TPartitionId partition_id,
         return status;
     }
 
+    if (tablet_txn_info->unique_key_merge_on_write && tablet_txn_info->partial_update_info &&
+        tablet_txn_info->partial_update_info->is_partial_update) {
+        status =
+                RowsetMetaManager::remove_partial_update_info(meta, tablet_id, rowset->rowset_id());
```

Comment:
should not remove here，it better to clear PartialUpdateInfo at the same time with the clear of rowset(RowsetMetaManager::remove)